arch/arm/core/aarch32/cortex_m/tz/secure_entry_functions.ld

```diff
@@ -7,9 +7,11 @@
 /* nRF-specific defines. */
 #if defined(CONFIG_CPU_HAS_NRF_IDAU) && CONFIG_ARM_NSC_REGION_BASE_ADDRESS == 0
 	/* This SOC needs the NSC region to be at the end of an SPU region. */
+	#define __NSC_ALIGN (ALIGN(CONFIG_NRF_SPU_FLASH_REGION_SIZE) \
+			- MAX(32, (1 << LOG2CEIL(__sg_size))))
 	#define NSC_ALIGN \
-		. = ALIGN(CONFIG_NRF_SPU_FLASH_REGION_SIZE) \
-			- (1 << LOG2CEIL(__sg_size))
+		. =  (__NSC_ALIGN + ((ABSOLUTE(.) > __NSC_ALIGN) \
+				? CONFIG_NRF_SPU_FLASH_REGION_SIZE : 0))
```

Comment:
Is it possible that even the next SPU boundary is still before the current location counter? I think no, due to the alignment done by __NCS_ALIGN, but please check.

Reply:
It really should not be possible, can you give more details about your concern?

__NSC_ALIGN goes forward to the next 0x8000 boundary, then backwards __sg_size (32-4096 bytes). Worst case is if we are at an address

    addr % 0x8000 == 0x7FFF

for example 0x2FFFF. __NSC_ALIGN will worst case go back to 0x2F000, in which case the ternary `if` should add 0x8000, which results in 0x37000 which is after 0x2FFFF, so it should be safe in every case.

Reply:
No concerns, really. I only wanted you to elaborate and confirm :)